src/PyDynamic/uncertainty/propagate_convolution.py

```diff
@@ -67,6 +76,14 @@ def convolve_unc(x1, U1, x2, U2, mode="full"):
         x1, x2 = x2, x1
         U1, U2 = U2, U1
 
+    # convert 1d array of standard uncertainties to covariance matrix
+    # squeeze() ensures proper execution on arrays with only one dimension of non-zero length
+    if isinstance(U1, np.ndarray) and len(U1.squeeze().shape) == 1:
+        U1 = np.diag(np.square(U1.squeeze()))
+
+    if isinstance(U2, np.ndarray) and len(U2.squeeze().shape) == 1:
+        U2 = np.diag(np.square(U2.squeeze()))
+
```

Comment:
@mgrub Would you mind, if I try to "clean" that with regard to Robert C. Martin's clean code along the lines of:
```python
U1, U2 = convert_std_uncs_2_cov_matrices(U1, U2)
```
to reduce mixing abstractions and redundancy further down?

Reply:
This now is dealt with in the following changes.